firefox-ios/Client/Frontend/PasswordGenerator/View/PasswordGeneratorHeaderView.swift

```diff
@@ -0,0 +1,75 @@
+    private enum UX {
+        static let headerIconLabelSpacing: CGFloat = 10
+        static let headerVerticalPadding: CGFloat = 8
+        static let headerImageHeight: CGFloat = 24
+    }
+
+    private var scaledHeaderImageSize = UIFontMetrics.default.scaledValue(for: UX.headerImageHeight)
+    private lazy var headerImageWidthConstraint = headerImageView.widthAnchor.constraint(
+        equalToConstant: scaledHeaderImageSize)
+    private lazy var headerImageHeightConstraint = headerImageView.heightAnchor.constraint(
+        equalToConstant: scaledHeaderImageSize)
+
+    private lazy var headerLabel: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.font = FXFontStyles.Bold.body.scaledFont()
+        label.text = .PasswordGenerator.Title
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.headerLabel
+        label.accessibilityTraits = .header
+    }
+
+    private lazy var headerImageView: UIImageView = .build { imageView in
+        imageView.image = UIImage(named: StandardImageIdentifiers.Large.login)?.withRenderingMode(.alwaysTemplate)
+        imageView.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.headerImage
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.header
+        setupLayout()
+}
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        self.addSubviews(headerImageView, headerLabel)
+
+        // Header elements layout
+        NSLayoutConstraint.activate([
+            headerImageView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            headerImageView.centerYAnchor.constraint(equalTo: headerLabel.centerYAnchor),
+            headerImageWidthConstraint,
+            headerImageHeightConstraint,
+            headerLabel.leadingAnchor.constraint(
+                equalTo: headerImageView.trailingAnchor,
+                constant: UX.headerIconLabelSpacing),
+            headerLabel.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            headerLabel.topAnchor.constraint(
+                equalTo: self.topAnchor,
+                constant: UX.headerVerticalPadding),
+            headerLabel.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+        ])
+    }
+
+    func applyTheme(theme: any Common.Theme) {
+        headerImageView.tintColor = theme.colors.iconPrimary
+        headerLabel.textColor = theme.colors.textPrimary
+    }
+
+    func applyDynamicFontChange() {
+        scaledHeaderImageSize = UIFontMetrics.default.scaledValue(for: UX.headerImageHeight)
+        headerImageWidthConstraint.constant = scaledHeaderImageSize
+        headerImageHeightConstraint.constant = scaledHeaderImageSize
+        headerLabel.font = FXFontStyles.Bold.body.scaledFont()
```

Comment:
Nit: The font doesn't have to be set again when the dynamic font size changes - it adjusts automatically.